src/overlays/actors/ovl_En_Ta/z_en_ta.c

```diff
@@ -941,192 +1016,204 @@ void func_80B161C0(EnTa* this, PlayState* play) {
 }
 
-void func_80B16608(EnTa* this, PlayState* play) {
+void EnTa_IdleAfterCuccoGameFinished(EnTa* this, PlayState* play) {
     if (Actor_ProcessTalkRequest(&this->actor, play)) {
         switch (this->actor.textId) {
-            case 0x2085:
-                this->actionFunc = func_80B161C0;
+            case 0x2085: // Retry?
+                this->actionFunc = EnTa_WaitForPlayCuccoGameResponse;
                 break;
-            case 0x2086:
-                this->actionFunc = func_80B162E8;
+            case 0x2086: // Initial win
+                this->actionFunc = EnTa_WaitForMarryMalonResponse;
                 break;
-            case 0x2088:
-                this->actionFunc = func_80B1642C;
+            case 0x2088: // Later win
+                this->actionFunc = EnTa_TalkAfterCuccoGameWon;
                 break;
         }
+        // Disable immediate automatic talk
         this->actor.flags &= ~ACTOR_FLAG_16;
     } else {
+        // Enable immediate automatic talk
         this->actor.flags |= ACTOR_FLAG_16;
         func_8002F2CC(&this->actor, play, 1000.0f);
     }
-    this->unk_2E0 |= 1;
+    this->stateFlags |= TALON_STATE_FLAG_TRACKING_PLAYER;
 }
 
-void func_80B166CC(EnTa* this) {
-    s16 temp_v0 = this->blinkTimer - 1;
+void EnTa_BlinkWaitUntilNext(EnTa* this) {
+    s16 blinkTimer = this->blinkTimer - 1;
 
-    if (temp_v0 != 0) {
-        this->blinkTimer = temp_v0;
+    if (blinkTimer != 0) {
+        this->blinkTimer = blinkTimer;
     } else {
-        this->unk_2B0 = func_80B16700;
+        this->blinkFunc = EnTa_BlinkAdvanceState;
     }
 }
 
-void func_80B16700(EnTa* this) {
+void EnTa_BlinkAdvanceState(EnTa* this) {
     s16 blinkTimer = this->blinkTimer - 1;
 
     if (blinkTimer != 0) {
         this->blinkTimer = blinkTimer;
     } else {
+        // Next towards closed eyes: open -> half, half -> closed
         s16 nextEyeIndex = this->eyeIndex + 1;
-        s16 blinkTimer = 3;
 
-        if (nextEyeIndex >= blinkTimer) {
-            this->eyeIndex = 0;
-            if (this->unk_2CE > 0) {
-                this->unk_2CE--;
+        // If the eyes were already closed, set the eyes open and set a new blink timer.
+        // If just woken up (rapidBlinks > 0), blink a few times
+        // in quick succession before starting the normal blink cycle.
+        if (nextEyeIndex >= TALON_EYE_INDEX_MAX) {
+            this->eyeIndex = TALON_EYE_INDEX_OPEN;
+            if (this->rapidBlinks > 0) {
+                this->rapidBlinks--;
                 blinkTimer = 1;
             } else {
                 blinkTimer = (s32)(Rand_ZeroOne() * 60.0f) + 20;
             }
             this->blinkTimer = blinkTimer;
-            this->unk_2B0 = func_80B166CC;
+            this->blinkFunc = EnTa_BlinkWaitUntilNext;
         } else {
             this->eyeIndex = nextEyeIndex;
             this->blinkTimer = 1;
         }
     }
 }
 
-void func_80B167C0(EnTa* this) {
+void EnTa_AnimLoopCurrent(EnTa* this) {
```

Comment:
Not sure about `Loop` since `Animation_PlayOnce` is what actually gets called. Perhaps `AnimPlayCurrent`, or possibly just `AnimNormal` to follow the style of the others where this is just the normal animation that doesn't do anything special.

Reply:
The point here was that when this is set as the current `animFunc`, current animation will be looped (`Animation_PlayOnce` is called again as soon as the animation ends). As opposed to `EnTa_AnimRunToEnd` that does nothing after the current animation finishes.

I think `PlayCurrent` or `Normal` would not describe this enough?

Reply:
I would use "repeat" or some term that gets that same meaning across without actually using the term "loop". Cause as mentioned, loop is a different type of playback method that is a part of the animation system (which they should have used here, but unfortunately did not)

Reply:
That makes sense, I'll change this to `repeat`